src/tapis-ui/components/files/FileListing/FileListing.tsx

```diff
@@ -58,15 +42,23 @@ const FileListing: React.FC<FileListingProps> = ({
     [onSelect]
   )
 
-  if (!result || result.loading) {
+  if (isLoading) {
     return <div className="file-list"><LoadingSpinner /></div>
   }
 
-  if (result.error) {
-    return <Message canDismiss={false} type="error" scope="inline">{result.error.message}</Message>
+  if (error) {
+    return <Message canDismiss={false} type="error" scope="inline">{(error as any).message}</Message>
   }
 
-  const files: Array<Files.FileInfo | null> = result.results;
+  const pages = data?.pages;
```

Comment:
Can this concatenation logic be brought into the `useList` hook? We'll probably have to reuse it in any component that does a file listing.